src/fsharp/CompileOps.fs

```diff
@@ -1700,6 +1700,7 @@ let DefaultReferencesForScriptsAndOutOfProjectSources(assumeDotNetFramework) =
           yield "System.Numerics" 
       else
           yield Path.Combine(Path.GetDirectoryName(typeof<System.Object>.Assembly.Location),"mscorlib.dll"); // mscorlib
+          yield typeof<System.Object>.Assembly.Location; // System.Private.CoreLib
```

Comment:
I don't, but as already mentioned in the original PR, there is a very unequivocal error message saying it's needed:
```
The type 'Object' is required here and is unavailable. You must add a reference to assembly System.Private.CoreLib
A reference to the type 'System.Object' in assembly 'mscorlib' was found, but the type could not be found in that assembly
```
Which kind of points to some issue in resolving forwarded types. Perhaps somebody from the dotnet/core team can shed a light?